libzecale/tests/aggregator/aggregator_dummy_test.cpp

```diff
@@ -82,118 +100,133 @@ void test_aggregator_with_batch(
     }
 }
 
-template<typename nsnarkT, typename wppT, typename wverifierT>
+template<typename wppT, typename wsnarkT, typename nverifierT, typename hashT>
 void test_aggregate_dummy_application()
 {
-    using nppT = other_curve<wppT>;
-    using wsnarkT = typename wverifierT::snark;
+    using npp = other_curve<wppT>;
+    using nsnark = typename nverifierT::snark;
 
     static const size_t batch_size = 2;
     static const size_t public_inputs_per_proof = 1;
```

Comment:
Why not add the size of the instance as part of another attribute of the application pool: https://github.com/clearmatics/zecale/blob/develop/libzecale/core/application_pool.hpp?
As such the new attribute `app_pool.nb_pub_inputs` (or however it is named) can be passed to the `aggregator` circuit to instantiate it's internal constants. That allows to check that the instance size of the extended proofs `push`'ed in the application pool equals `app_pool.nb_pub_inputs` and enables to fail early (before starting the assignment of circuit wires)

Reply:
Good catch.  I've added it to aggregator server for now, since that's where most of these checks and error handling is currently done (I added a bunch in a previoius PR, but didn't add this one).
Eventually it would be great to include the number of inputs in the proof / vk types so that these errors are caught at compile time / when the tx is decoded, but that might be a more fundamental change.

Reply:
> Good catch. I've added it to aggregator server for now, since that's where most of these checks and error handling is currently done (I added a bunch in a previoius PR, but didn't add this one).

That's fine for now to keep this on the server. Nevertheless, that shouldn't be where this value is set in the long run. The "uncompressed" instance size is function of the underlying base application and so checking received instance size outside of the scope of the application doesn't really make sense and affects the "genericity" of the Zecale server. The base app "compressed" instance size (the output of the hash function - see "GGPR's trick" - used on the base app instance) is a Zecale protocol param however, and so that's fine to set this on the aggregator server.

Reply:
Right.  Eventually it seems we may be able to support a per-application number of inputs (up to some predetermined max).  Note, it would not really be feasible to keep this as a variable yet anyway - we need a generic `snarkT::verification_key_get_num_inputs()` method first.  clearmatics/zeth#291